compiler/noirc_frontend/src/parser/parser/function.rs

```diff
@@ -76,16 +79,32 @@ fn function_modifiers() -> impl NoirParser<(bool, ItemVisibility, bool)> {
         })
 }
 
+pub(super) fn numeric_generic() -> impl NoirParser<UnresolvedGeneric> {
+    keyword(Keyword::Let)
+        .ignore_then(ident())
+        .then_ignore(just(Token::Colon))
+        .then(parse_type())
+        .map(|(ident, typ)| UnresolvedGeneric::Numeric { ident, typ })
+}
+
+pub(super) fn generic_variable() -> impl NoirParser<UnresolvedGeneric> {
+    ident().map(UnresolvedGeneric::Variable)
+}
+
+pub(super) fn generic() -> impl NoirParser<UnresolvedGeneric> {
+    numeric_generic().or(generic_variable())
```

Comment:
```suggestion
    generic_variable().or(numeric_generic())
```
Even more of a nitpick: these parsers are both mutually exclusive so we should put the more common case first to parse a bit faster.